### DISCUSSION TITLE
Customising the download url

### DISCUSSION THREAD

**Discussion by jagjit-arora** [flapdoodle-oss/de.flapdoodle.embed.mongo#487] [Q&A] [answered]
How can I customize the whole url to download binaries based on operating system? I want to provide it different url for windows and Linux  for the binaries

**Comment by michaelmosmann** [comment]
@jagjit-arora you have to change this property: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/0148baff2e043cbb77d5de352a0fa35087862372/src/main/java/de/flapdoodle/embed/mongo/transitions/PackageOfCommandDistribution.java#L47 and implement your own.. 

may i ask what kind of problem you try to solve?

**Reply by jagjit-arora** [reply]
When working locally on windows the URL for binary I am trying to fetch is totally different from the URL in Linux used in the pipeline job runner. Implementing my own package resolver helped me to tackle the issue.

**Reply by michaelmosmann** [reply]
@jagjit-arora is this of any use for other users? should the existing implementation should be improved?

**Reply by jagjit-arora** [reply]
I do think that people sitting behind a firewall or have their own distribution binaries can use this. What are your thoughts on this?

**Reply by michaelmosmann** [reply]
@jagjit-arora I think there are some people out there with this kind of problem.. so i think it is useful:)

**Reply by jagjit-arora** [reply]
Definitely.

**Reply by michaelmosmann** [reply, marked as answer]
@jagjit-arora .. please have a look: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/main/docs/Customizations.md#customize-package-resolver

**Reply by jagjit-arora** [reply]
Awesome, this helps to customize the download url as a whole and based on runtime environment along with the command that is to be executed.

**Reply by michaelmosmann** [reply]
@jagjit-arora your welcome;)
